src/main/scala/io/iohk/ethereum/ledger/InMemoryWorldStateProxy.scala

```diff
@@ -155,7 +155,8 @@ class InMemoryWorldStateProxy private(
     copyWith(accountsStateTrie = accountsStateTrie.put(address.bytes, account))
 
   override def deleteAccount(address: Address): InMemoryWorldStateProxy =
-    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes))
+    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes),
+      contractStorages = contractStorages - address)
```

Comment:
back to accountCodes
so we should add `accountCodes = accountCodes - address` ?
and have
```
    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes),
      contractStorages = contractStorages - address, accountCodes = accountCodes - address)
```